proposals/0123-block-revenue-distribution.md

```diff
@@ -137,46 +137,64 @@ vote state field `pending_delegator_rewards` and added to the balance of vote
 account. If either of these conditions is violated, the delegator rewards amount
 MUST be burned.
 
-### Runtime: Delegator Rewards Distribution
+### Runtime: Block Revenue Distribution
 
-When calculating stake delegation rewards for a particular completed reward
-epoch, construct a list of all vote accounts that were initialized at the
-beginning of the reward epoch and had a non-zero active stake delegation. For
-each vote account, retrieve its state at the end of the reward epoch and check
-the `pending_delegator_rewards` field in its vote state. Let this value be `P`.
-If `P` is non-zero, use it to calculate rewards for each of the stake accounts
-delegated to the vote account as follows: 
+At the beginning of each epoch, pending delegator rewards will be collected from
+actively staked vote accounts into the epoch rewards sysvar account and then
+distributed to stake accounts utilizing the existing partitioned rewards
+distribution mechanism described in [SIMD-0118].
 
-1. Sum all active stake delegated to the vote account during the reward epoch
-epoch. Let this total be `A`.
+#### Delegator Rewards Calculation
 
-2. For each individual stake account, multiply its active stake from the
-reward epoch by `P`, and divide the result by `A` using integer division.
-Discard any fractional lamports.
+Delegator rewards MUST be calculated during both the beginning of the first
+block of an epoch `N` and after restarting during partitioned rewards
+distribution.
 
-After calculating all individual stake rewards, sum them to obtain `D`, the
-total distribution amount. Because of integer division, the full amount `P` may
-not be distributed so compute the amount to be burned, `B`, as the difference
-between `P` and `D`.
+For each vote account, get its total active stake delegation
+during the reward epoch `N - 1`. Let this value be `A`. If `A` is zero, skip
+this vote account (any pending delegator rewards will be distributed in a future
+epoch when the vote account has active stake.)
 
-If no blocks in the epoch following the completed reward epoch have been
-processed yet, subtract `B` from both the vote account’s lamport balance and its
-`pending_delegator_rewards` field and store the updated vote account. Finally,
-the burn amount `B` should also be deducted from the cluster capitalization.
+Then for each vote account, get its pending delegator rewards from the
+`pending_delegator_rewards` field in the vote state at the end of reward epoch
+`N - 1`. Let this value be `P`. If `P` is zero, skip this vote account as there
+is nothing to be distributed.
+
+Lastly, if this is the first block of epoch `N`, the vote state's
+`pending_delegator_rewards` field MUST be reset to `0` and `P` lamports MUST be
+deducted from the vote account's lamport balance and credited to the epoch
```

Comment:
how will these debits be recorded on chain?

Reply:
Good point. Though I don't think they should be recorded onchain, we should record these debits in the block rewards field as we do for other balance changes (rewards, fee revenue, rent due, etc). I don't think this needs to be described in the SIMD

Reply:
are those fields only under consensus implicitly due to their denoting state updates? for some reason i thought that they were explicit

Reply:
ah the glory of being primary implementer